soh/src/overlays/misc/ovl_kaleido_scope/z_kaleido_item.c

```diff
@@ -353,40 +353,23 @@ void KaleidoScope_DrawItemSelect(PlayState* play) {
                 KaleidoScope_SetCursorVtx(pauseCtx, index, pauseCtx->itemVtx);
 
                 if ((pauseCtx->debugState == 0) && (pauseCtx->state == 6) && (pauseCtx->unk_1E4 == 0)) {
-                    // only allow mask select when:
-                    // the shop is open:
-                    // * zelda's letter check: gSaveContext.eventChkInf[4] & 1
-                    // * kak gate check: gSaveContext.infTable[7] & 0x40
-                    // and the mask quest is complete: gSaveContext.eventChkInf[8] & 0x8000
-                    if (CVarGetInteger("gMaskSelect", 0) &&
-                        (gSaveContext.eventChkInf[8] & 0x8000) &&
-                        cursorSlot == SLOT_TRADE_CHILD && CHECK_BTN_ALL(input->press.button, BTN_A) &&
-                        (gSaveContext.eventChkInf[4] & 1) &&
-                        (gSaveContext.infTable[7] & 0x40)) {
+                    if ((CVarGetInteger("gMaskSelect", 0) || gSaveContext.n64ddFlag) && 
```

Comment:
is there a way to do this without always having it enabled for rando? 

i know the routing issues (such as it being faster to get all masks and then go deku theater with mask select, vs needing to go back to shop without)) are being handled in GetNextChildTradeItem, but it feels a little odd to be able to enter mask select mode when there's nothing to select.

i feel like a naive solution would be

```suggestion
                    if ((CVarGetInteger("gMaskSelect", 0) || (gSaveContext.n64ddFlag && Randomizer_GetSettingValue(RSK_STARTING_BUNNY_HOOD))) && 
```
but maybe something more robust could look like moving `possibleItems` out of `GetNextChildTradeItem` and splitting it up to have `UpdatePossibleChildTradeItems` called as part of `GetNextChildTradeItem` as well as a new `HasMultipleChildTradeItems` method that could be used here. That might also have the benefit of not needing to rebuild that vector every time we switch items (but that part would be beyond the scope of this)

Reply:
It's applicable outside of start-with-bunnyhood to account for https://github.com/HarbourMasters/Shipwright/issues/1686. Your idea of splitting it up is still valid, though maybe as another tech debt item? Or should it block this?

Reply:
how about making `CanEnterMaskSelect()` or something, having it return `CVarGetInteger("gMaskSelect", 0) || gSaveContext.n64ddFlag`, adding a todo in there (could be as simple as `todo: don't just always enable in rando`), and then updating this line to be
```suggestion
                    if (CanEnterMaskSelect() && 
```